### PR TITLE
P-315 add lit staking vc

### PR DESCRIPTION
**Pull request by outofxxx**
LIT staking credential: [interface & example](https://www.notion.so/web3builders/NewYear-LIT-staking-7d59e8b6beae4a7cab5da55baf584953)

### CODE REVIEW

**Review by BillyWooo — COMMENTED** [review]
"[primitives/staking/Cargo.toml](https://github.com/litentry/litentry-parachain/pull/2332/files#diff-4716a35a260db2e93650ff0390cc0d880b691e0f13fc9f89767d1e4ee3c1b9d8)" maybe it's not necessary.

missing cli-client defines update.

File: tee-worker/litentry/core/credentials/src/litentry_profile/lit_staking.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	pub amount: u128,
+}
+
+impl LITStakingAmount {
+	pub fn new(amount: u128) -> Self {
+		Self { amount }
+	}
+}
+
+impl RangeCredentialDetail for LITStakingAmount {
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/credentials/src/litentry_profile/lit_staking.rs:38]: It's great that we have this trait :)

Maybe consider moving it to a common place

File: tee-worker/litentry/core/assertion-build/src/lit_staking.rs [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+		client: &mut LitentryStakingClient,
+		identities: &[Identity],
+	) -> Result<u128> {
+		let mut total_staking_amount = 0_u128;
+
+		for identity in identities {
+			let storage_key = DelegatorState::delegator_state_storage_key(identity)?;
+			let storage_in_hex = client.query_delegator_state(&storage_key)?;
+
+			let delegator = DelegatorState::decode_delegator(&storage_in_hex)?;
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/assertion-build/src/lit_staking.rs:129]: `and_then` might be of better readability here?

File: tee-worker/litentry/core/assertion-build/src/lit_staking.rs [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+		let params = "0xa686a3043d0adcf2fa655e57bc595a78131da8bc800de21b19b3ba9ed33cfacc";
+		let acc = identity
+			.to_account_id()
+			.ok_or(Error::RequestVCFailed(Assertion::LITStaking, ErrorDetail::ParseError))?;
+		let cocat = Twox64Concat::hash(acc.as_ref());
+
+		Ok(params.to_string() + &hex::encode(&cocat))
+	}
+
+	fn decode_delegator(storage_in_hex: &str) -> Result<Delegator<AccountId, Balance>> {
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/assertion-build/src/lit_staking.rs]: We'll need to double check if `Balance` is parentchain balance, even if they eventually use the same type, it's always better to make it explicit.

File: tee-worker/litentry/core/assertion-build/src/lit_staking.rs [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+					)
+				})?
+				.map_err(|e| {
+					Error::RequestVCFailed(
+						Assertion::LITStaking,
+						ErrorDetail::StfError(ErrorString::truncate_from(
+							format!("{:?}", e).as_bytes().to_vec(),
+						)),
+					)
+				})
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/assertion-build/src/lit_staking.rs]: same here - can we simplify this

**Review by Kailai-Wang — COMMENTED** [review]
In general looks good!

File: tee-worker/litentry/core/credentials/src/litentry_profile/lit_staking.rs (hunk 1 above)
Comment (BillyWooo) [reply tee-worker/litentry/core/credentials/src/litentry_profile/lit_staking.rs:38]: Yes. We had discussed this to do a refactor about these kind of usage/trait.

File: tee-worker/litentry/core/assertion-build/src/lit_staking.rs (hunk 3 above)
Comment (outofxxx) [reply tee-worker/litentry/core/assertion-build/src/lit_staking.rs]: using ParentchainBalance instead.

File: tee-worker/litentry/core/assertion-build/src/lit_staking.rs (hunk 2 above)
Comment (outofxxx) [reply tee-worker/litentry/core/assertion-build/src/lit_staking.rs:129]: Actually, I prefer the approach of `?` in this place

**Review by BillyWooo — APPROVED** [review]
LGTM

**Pull request merged**